song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceImpl.java

```diff
@@ -208,6 +218,63 @@ public void updateAnalysis(
     return analyses;
   }
 
+  @Override
+  public GetAnalysisResponse getAnalysis(
+      @NonNull String studyId,
+      @NonNull Set<String> analysisStates,
+      @NonNull int page,
+      @NonNull int size) {
+    studyService.checkStudyExist(studyId);
+    resolveSelectedAnalysisStates(analysisStates);
+
+    val sort = new Sort(Sort.Direction.ASC, ANALYSIS_ID);
+    val pageRequest = PageRequest.of(page, size, sort);
+
+    val spec =
+        new Specification<Analysis>() {
+          @Override
+          public Predicate toPredicate(
+              Root<Analysis> root, CriteriaQuery<?> query, CriteriaBuilder criteriaBuilder) {
+            // The reason for having this check is because of the issue: https://github.com/spring-projects/spring-data-jpa/issues/532
+            // in order to avoid the hibernate error and perform a fetch join, do not apply fetch join criteria
+            // on count query. If the result type is long, that means a count query is fired.
+            if (!query.getResultType().equals(Long.class)) {
+              root.fetch(ANALYSIS_SCHEMA, LEFT);
+              root.fetch(ANALYSIS_DATA, LEFT);
+            }
+            query.distinct(true);
+            return criteriaBuilder.and(
+                equalsStudyPredicate(root, criteriaBuilder, studyId),
+                whereStatesInPredicate(root, analysisStates));
+          }
+        };
+
+    val analyses = repository.findAll(spec, pageRequest);
+
+    analyses.forEach(
+        a -> {
+          val id = a.getAnalysisId();
+          a.setFiles(unsecuredReadFiles(id));
+          a.setSamples(readSamples(id));
+          a.populatePublishTimes();
```

Comment:
Just want to double check, it looks like this pr is just to add paging not to fix the in memory joins?

Reply:
just adding paging, there is another ticket for investigating how to improve performance.